pkg/ccl/backupccl/testdata/backup-restore/backup-permissions

```diff
@@ -0,0 +1,118 @@
+# Test permissions checks for non-admin users running BACKUP.
+new-server name=s1
+----
+
+exec-sql
+CREATE DATABASE d;
+CREATE TABLE d.t (x INT);
+INSERT INTO d.t VALUES (1), (2), (3);
+----
+
+# Cluster backup should succeed as a root user.
+exec-sql
+BACKUP TO 'nodelocal://0/test-root/'
+----
+
+# Backups should succeed as a non-root user with admin role.
+exec-sql
+CREATE USER testuser;
+GRANT ADMIN TO testuser;
+----
+
+exec-sql user=testuser
+BACKUP TO 'nodelocal://0/test-nonroot-cluster';
+BACKUP DATABASE d TO 'nodelocal://0/test-nonroot-db';
+BACKUP TABLE d.t TO 'nodelocal://0/test-nonroot-table';
+----
+
+# Start a new cluster with the same IO dir.
+new-server name=s2 allow-implicit-access
+----
+
+exec-sql server=s2
```

Comment:
nit: It looks like `exec-sql` uses the last created server, so we don't need to specify `server=s2` and `server=s3` on all of the following exec-sql commands.

Reply:
done!